### ISSUE TITLE
evaluating with trainer.py with TPU results in sudden RAM spike and crash

### ISSUE THREAD

**Issue by LeonieWeissweiler** [huggingface/transformers#4991]
# 🐛 Bug

## Information

Model I am using (Bert, XLNet ...): Roberta with the official how to train from scratch example

Language I am using the model on (English, Chinese ...): Esperanto

The problem arises when using:
* [ ] the official example scripts: (give details below)
* [x] my own modified scripts: (give details below)

## To reproduce

1. follow the official how to train with esperanto example
2. define a second dataset for testing, both datasets are TextDatasets, the train set is around 300MB, the test set around 3MB
3. run trainer.train() with evaluate_during_training

The evaluation loop runs a few examples before crashing the colab session due to an unknown reason. I have managed to increase the number of examples from 2 to 6 by reducing the per_device_eval_batch_size. My eval dataset only contains 31 examples. When checking the colab logs after crashing, it seems that trainer.py tried to allocate 4GB of memory before crashing, which seems unrealistic to me given the size of the datasets.

Specifically (I think this is the relevant part, please correct me if I'm wrong):

2020-06-14 21:45:53.187694: E tensorflow/compiler/xla/xla_client/xla_util.cc:76] (1) Resource exhausted: Attempting to reserve 2.78G at the bottom of memory. That was not possible. There are 2.75G free, 0B reserved, and 2.75G reservable.

## Expected behavior

I expected the eval loss or error to be computed. 

## Environment info

     
- `transformers` version: 2.11.0
- Platform: colab
- Python version: 3.6
- Using GPU in script?: no, TPU with xla
- Using distributed or parallel set-up in script?: in theory, but only 1 TPU core available

**Comment by finiteautomata** [comment]
Any update on this? I've got a similar issue
